# Rule

<!-- end auto-generated rule header -->

<!-- TODO -->

## Options

| Name | Description | value |
|:-----|:------------|:------|

<!-- TODO -->

### Defaults

<!-- TODO -->
This is the default values when the rule is enabled:

```json
{}
```

### Usage

<details>
<summary>Usage of <code>option</code></summary>

#### Configuration

```json
{}
```

#### ❌ Invalid

```
```

#### ✅ Valid

```
```

</details>

<br />
